docs/src/developer/guide/architecture.md

```diff
@@ -1,10 +1,42 @@
 # System Architecture
-<!-- Replace all of the titles with relevant titles -->
+In this section, we outline the system architecture using ER Diagrams, Software Component Diagrams etc. and key libraries / frameworks used in this project.
 
 ## Frameworks Used
 
+The following is a list, with brief descriptions, of the key components used in creating this platform. Please refer to their individual documentation for in-depth technical information.
+
+| Logo | Name | Notes |
+|------------|---------|----------------|
+|![Django](img/arch-framework-django-1.png){: style="height:30px;width:30px"} | [Django](https://djangoproject.com) | Django makes it easier to build better web apps more quickly and with less code. | 
```

Comment:
Please update the image names to something like:

```suggestion
|![Django](img/architecture-django-1.png){: style="height:30px;width:30px"} | [Django](https://djangoproject.com) | Django makes it easier to build better web apps more quickly and with less code. | 
```

And remove any images you are no longer using